h2o-core/src/main/java/water/init/NetworkInit.java

```diff
@@ -185,6 +197,59 @@ public static void initializeNetworkSockets( ) {
     H2O.CLOUD_MULTICAST_PORT = NetworkUtils.getMulticastPort(hash);
   }
 
+  public static H2OHttpConfig webServerParams(H2O.OptArgs args) {
+    final H2OHttpConfig params = new H2OHttpConfig();
+    params.jks = args.jks;
+    params.jks_pass = args.jks_pass;
+    params.loginType = parseLoginType(args);
+    params.login_conf = args.login_conf;
+    params.form_auth = args.form_auth;
+    params.session_timeout = args.session_timeout;
+    params.user_name = args.user_name;
+    params.port = args.port;
+    params.baseport = args.baseport;
+    params.web_ip = args.web_ip;
+    params.ip = args.ip;
+    params.network = args.network;
+    params.context_path = args.context_path;
+
+    params.h2oExtendedHeaders = new LinkedHashMap<>();
+    params.h2oExtendedHeaders.put("X-h2o-build-project-version", H2O.ABV.projectVersion());
+    params.h2oExtendedHeaders.put("X-h2o-rest-api-version-max", Integer.toString(water.api.RequestServer.H2O_REST_API_VERSION));
+    params.h2oExtendedHeaders.put("X-h2o-cluster-id", Long.toString(H2O.CLUSTER_ID));
+    params.h2oExtendedHeaders.put("X-h2o-cluster-good", Boolean.toString(H2O.CLOUD.healthy()));
```

Comment:
Cluster health is not a static property, it changes in time. Cluster can become unhealthy and can resume normal operation after a while.